### PR TITLE
CANS-335 Restriction Access Drop Down

### PR DESCRIPTION
**Pull request by vladimirPe**
## Description
Implementation of a dropdown field on the add/edit child page.
https://osi-cwds.atlassian.net/browse/CANS-335

## Tests

- [x] I used TDD to develop the feature
- [x] I have included unit tests
- [ ] I have included new acceptance tests or modified existing ones
- [ ] I have included other tests

## Types of changes

- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (No behavioral changes)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] I ran all my tests locally which were green.
- [ ] My code adds no new issues to Code Climate
- [x] I ran all the linters for the project.
- [x] My code is in a stable state ready to be deployable, but not necessarily complete.
- [x] I promise on my honor as a CWDS developer to ensure I don't break the build (or bring donuts if I leave things broken), to check linters, use best practices, to leave the code in better shape than I found it, and to have a good day.

### CODE REVIEW

File: app/javascript/Application/components/Client/ClientAddEditForm.js [hunk 1]
```diff
@@ -145,6 +153,24 @@ class ClientAddEditForm extends Component {
...
+          if (!sensitivityTypes || sensitivityTypes.length === 0) {
+            this.setState({
+              childInfo: {
+                ...this.state.childInfo,
+                sensitivity_type: null,
+              },
+            });
+          }
+        })
+        .catch(() => {});
```
[the remark this answers is not in the document]
Comment (vladimirPe) [reply app/javascript/Application/components/Client/ClientAddEditForm.js]: `#fetch` is  firing a global message so may be we  don't need catch at all here

File: app/javascript/Application/components/Client/ClientAddEditForm.js [hunk 2]
```diff
@@ -111,6 +112,7 @@ const prepareChildInfoValidation = (childInfo, isNewForm) => {
     dob: !isNewForm,
     external_id: !isNewForm,
     county: !isNewForm,
+    sensitivity_type: true,
```
Comment (denys-davydov) [inline app/javascript/Application/components/Client/ClientAddEditForm.js]: This field becomes redundant when no validation in place

Comment (vladimirPe) [reply app/javascript/Application/components/Client/ClientAddEditForm.js]: good catch

**Comment by vladimirPe** [comment]
Jenkins retest this please

**Pull request merged**